python/lsst/meas/modelfit/display/cModelDisplay.py

```diff
@@ -0,0 +1,264 @@
+        Multi-component shapelet model of the dev component of CModel jointly
+        fit with the exp component
+    expJointShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model of the exp component of Cmodel jointly
+        fit with the dec component
+    fitSysToMeasSys.flux : 'double'
+        Reference flux used to construct coordinate transformation between
+        the pixel flux system and the model flux system
+    """
+
+    # build a unit system transformation object
+    center = record.getCentroid()
+    position = exposure.getWcs().pixelToSky(center)
+    measSys = measMod.UnitSystem(exposure)
+    approxFlux = record.get("base_PsfFlux_flux")
+    fitSys = measMod.UnitSystem(position, exposure.getCalib(), approxFlux)
+    fitSysToMeasSys = measMod.LocalUnitTransform(position, fitSys, measSys)
+
+    # Build the Shapelet objects
+    ctrl = cModelConfig.makeControl()
+    baseKey = "modelfit_CModel"
+    nonlinearKeys = ["{}_{{model}}_nonlinear_{p}".format(baseKey, p=p)
+                     for p in range(3)]
+    fixedKeys = ["{}_{{model}}_fixed_{p}".format(baseKey, p=p)
+                 for p in range(2)]
+    fluxKey = "{}_{{model}}_flux".format(baseKey)
+
+    # Get parameters for the dev model
+    devNonLinearParams = np.array([record.get(key.format(model="dev"))
+                                  for key in nonlinearKeys])
+    devFixedParams = np.array([record.get(key.format(model="dev"))
+                              for key in fixedKeys])
+    devAmp = np.array([record.get(fluxKey.format(model="dev"))])
+    devAmp /= fitSysToMeasSys.flux
+    devShapelet = ctrl.dev.getModel().makeShapeletFunction(devNonLinearParams,
+                                                           devAmp,
+                                                           devFixedParams)
+    devShapelet.transformInPlace(fitSysToMeasSys.geometric)
+
+    # Get parameters for the exp model
+    expNonLinearParams = np.array([record.get(key.format(model="exp"))
+                                  for key in nonlinearKeys])
+    expFixedParams = np.array([record.get(key.format(model="exp"))
+                              for key in fixedKeys])
+    expAmp = np.array([record.get(fluxKey.format(model="exp"))])
+    expAmp /= fitSysToMeasSys.flux
+    expShapelet = ctrl.exp.getModel().makeShapeletFunction(expNonLinearParams,
+                                                           expAmp,
+                                                           expFixedParams)
+    expShapelet.transformInPlace(fitSysToMeasSys.geometric)
+
+    # Get joint shapelet model
+    fracDev = record.get("{}_fracDev".format(baseKey))
+    jointFlux = np.array([record.get("{}_flux".format(baseKey))])
+    jointFlux /= fitSysToMeasSys.flux
+    devJointShapelet = ctrl.dev.getModel()\
+        .makeShapeletFunction(devNonLinearParams, jointFlux*fracDev,
+                              devFixedParams)
```

Comment:
Should be possible to fit this on one (110-char) line.  Note that the 80-char rule is not operative (RFC was accepted but not implemented).